File: Client/Gamify.Client.Net/Gamify.Client.Net/Services/IGameService.cs
using Gamify.Contracts.Notifications;
using Gamify.Contracts.Requests;

namespace Gamify.Client.Net.Services
{
    public interface IGameService<TRequest, UNotification> : IGameSender<TRequest>, IGameListener<UNotification>
        where TRequest : IRequestObject
        where UNotification : INotificationObject
    {
    }
}

File: Client/Java/gamify-java-client/src/main/java/ar/com/mdf/gamify/services/IGameService.java
package ar.com.mdf.gamify.services;

import ar.com.mdf.gamify.objects.notification.INotificationObject;
import ar.com.mdf.gamify.objects.request.IRequestObject;
import ar.com.mdf.gamify.observers.IGameObserver;

public interface IGameService<T extends IRequestObject, U extends INotificationObject > extends IGameSender<T>, IGameListener<U>{
	
		public void suscribe(IGameObserver<U> observer);
		public void send(T request);
		
}
